crates/daemon/src/monitor.rs

```diff
@@ -466,25 +470,76 @@ impl Actor {
 
         tracing::trace!("Updating status of {num_transactions} transactions",);
 
-        let histories = self
-            .client
-            .batch_script_get_history(self.state.monitoring_scripts())
-            .context("Failed to get script histories")?;
-
-        let mut ready_events = self.state.update(
-            latest_block_height,
-            histories
-                .into_iter()
-                .map(|list| {
-                    list.into_iter()
-                        .map(|response| TxStatus {
-                            height: response.height,
-                            tx_hash: response.tx_hash,
-                        })
-                        .collect()
-                })
-                .collect(),
-        );
+        // TODO: Evaluate buffer; if we do batches of 25 and allow buffer of 100 here, does that
+        // mean we can at most process 4 batches at the time?  What happens if the buffer is
+        // exhausted? (I would expect the next messages will be queued) We use tokio::sync::
+        // mpsc::channel because we need the receiver to be in an async context because we do async
+        // work upon receiving script updates
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(100);
+
+        let scripts = self
+            .state
+            .monitoring_scripts()
+            .cloned()
+            .collect::<Vec<Script>>();
+
+        let batches = scripts.chunks(BATCH_SIZE).map(|batch| batch.to_owned());
+
+        let client = self.client.clone();
+
+        let batch_futures = batches.into_iter().map(move |batch| {
+            tokio::task::spawn_blocking({
+                let client = client.clone();
+                let sender = sender.clone();
+                move || {
+                    for script in batch {
+                        // TODO: Evaluate if we should notify the receivers about errors, I feel
```

Comment:
Agree, as long as we don't intend to deal with the errors there is no need in bubbling it up

Reply:
there are some errors that we might be interested in, e.g. if we're exceeding the rate, we could slow down the requests.

Reply:
> there are some errors that we might be interested in, e.g. if we're exceeding the rate, we could slow down the requests.

Is it fair to say that we can add such a feature later, once we know how this performs on testnet? 
I find it hard to design this without knowing how the solution will perform under load.

We might also want to change the `BATCH_SIZE` - 25 is a somewhat arbitrary number. At the moment we process batches of 25 requests in parallel, for `7000` subscriptions this means `280` worker threads in parallel. When running the test that simulates this behavior this did not cause problems on our local machines, but there was CPU load. We will have to see how this performs on the testnet machine.